android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -442,7 +443,8 @@ protected void finishSharedAnimation(int tag) {
           if (key.equals(Snapshot.TRANSFORM_MATRIX)) {
             preparedValues.put(key, value);
           } else {
-            preparedValues.put(key, (double) PixelUtil.toDIPFromPixel((int) value));
+            float pixelsValue = Utils.convertToFloat(value);
+            preparedValues.put(key, (double) PixelUtil.toDIPFromPixel(pixelsValue));
```

Comment:
```suggestion
            float dipValue = PixelUtil.toDIPFromPixel(pixelsValue);
            preparedValues.put(key, (double) dipValue);
```
BTW why do we need to cast it to double?